pkg/handler/collector/s3/bucket/bucket.go

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bucket
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+type BuildBucket interface {
+	GetDownloader(hostname string, port string, region string) Bucket
+}
+
+type BucketBuilder struct {
+}
+
+func (bd BucketBuilder) GetBucket(hostname string, port string, region string) Bucket {
+	return S3Bucket{
+		hostname,
+		port,
+		region,
+	}
+}
+
+type Bucket interface {
+	DownloadFile(ctx context.Context, bucket string, item string) ([]byte, error)
+	GetEncoding(ctx context.Context, bucket string, item string) (string, error)
+}
+
+type S3Bucket struct {
```

Comment:
it seems like this struct can be private since its public interface since its type is never returned